grails-app/services/org/pih/warehouse/order/OrderService.groovy

```diff
@@ -1101,4 +1101,30 @@ class OrderService {
         order.save(flush:true)
     }
 
+    /**
+     * Gets map of derived status for orders (Order id as a key and derived status as a value)
+     * Done to improve performance of getting orders derived statuses on order list page
+     * */
+    def getOrdersDerivedStatus(List orderIds) {
+        if (!orderIds) {
+            return [:]
+        }
+
+        def g = grailsApplication.mainContext.getBean('org.codehaus.groovy.grails.plugins.web.taglib.ApplicationTagLib')
+        def orderSummaryList =  OrderSummary.findAllByIdInList(orderIds)
```

Comment:
I assume this method is going to be slow due to the use of the order summary. Are we in for a rude awakening come testing? Or do you think this will perform ok?